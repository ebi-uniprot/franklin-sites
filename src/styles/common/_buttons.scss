@import '../settings';
@import '../colours';

.button {
  --main-color: var(--main-button-color, #{$colour-sea-blue});
  white-space: nowrap;

  user-select: none;

  svg {
    width: 0.9rem;
    height: 0.9rem;
    margin-top: -0.2rem;
    margin-right: 0.5rem;
    vertical-align: middle;
    fill: $colour-sky-white;
  }

  &.large-icon svg {
    width: 1.2rem;
    height: 1.2rem;
    margin-top: -0.2rem;
    margin-right: 0.5rem;
  }

  &.primary {
    &,
    &.disabled {
      background-color: $colour-sea-blue;
      background-color: var(--main-color);
    }

    &:hover,
    &:focus,
    &:focus-within {
      background-color: $colour-sea-blue;
      background-color: var(--main-color);
      filter: brightness(90%);
    }
  }

  &.secondary {
    &,
    &.disabled {
      color: $colour-sea-blue;
      color: var(--main-color);
      border: 0.1rem solid currentColor;
      font-weight: 700;

      svg {
        fill: currentColor;
      }
    }
  }

  &.tertiary {
    background: none;
    color: $colour-sea-blue;
    color: var(--main-color);
    font-weight: $global-weight-bold;
    padding: 0;
    line-height: 1.6;
    text-align: left;

    &:hover:disabled {
      color: $colour-sea-blue;
      color: var(--main-color);
      background-color: $colour-sky-white;
    }

    svg {
      fill: currentColor;
    }

    &.tertiary-icon__active {
      svg {
        fill: $colour-platinum;
      }
    }
  }
}

/* overriding Foundation */
.button-group {
  align-items: baseline;
  margin-bottom: $global-margin * 0.5;

  .tertiary {
    margin-right: 1rem;
  }
}
