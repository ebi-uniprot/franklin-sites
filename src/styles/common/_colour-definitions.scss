@import '../colours';

// PRIMARY
.colour-sapphire-blue {
  background: $colour-sapphire-blue;
  color: $colour-sky-white;
}
.colour-sea-blue {
  background: $colour-sea-blue;
  color: $colour-sky-white;
}
.colour-vivid-cerulean {
  background: $colour-vivid-cerulean;
  color: $colour-sky-white;
}
.colour-medium-turquoise {
  background: $colour-medium-turquoise;
  color: $colour-yankees-blue;
}
.colour-gainsborough {
  background: $colour-gainsborough;
  color: $colour-yankees-blue;
}

// GREY
.colour-yankees-blue {
  background: $colour-yankees-blue;
  color: $colour-sky-white;
}
.colour-independence {
  background: $colour-independence;
  color: $colour-sky-white;
}
.colour-weldon-blue {
  background: $colour-weldon-blue;
  color: $colour-sky-white;
}
.colour-pastel-blue {
  background: $colour-pastel-blue;
  color: $colour-yankees-blue;
}
.colour-platinum {
  background: $colour-platinum;
  color: $colour-yankees-blue;
}
.colour-sky-white {
  background: $colour-sky-white;
  color: $colour-yankees-blue;
}
// CURATION
.colour-reviewed {
  background: $colour-reviewed;
  color: $colour-yankees-blue;
}
.colour-unreviewed {
  background: $colour-unreviewed;
  color: $colour-yankees-blue;
}
// NAMESPACE
.colour-uniref {
  background: $colour-uniref;
  color: $colour-yankees-blue;
}
.colour-uniparc {
  background: $colour-uniparc;
  color: $colour-yankees-blue;
}
.colour-proteomes {
  background: $colour-proteomes;
  color: $colour-sky-white;
}
.colour-peptide-search {
  background: $colour-peptide-search;
  color: $colour-sky-white;
}
.colour-id-mapping {
  background: $colour-id-mapping;
  color: $colour-sky-white;
}
.colour-blast {
  background: $colour-blast;
  color: $colour-sky-white;
}
.colour-align {
  background: $colour-align;
  color: $colour-sky-white;
}

// Help section
.colour-help-green {
  background: $colour-help-green;
  color: $colour-sky-white;
}

// svg fills
.svg-colour-reviewed svg {
  color: $colour-reviewed;
}
.svg-colour-unreviewed svg {
  color: $colour-unreviewed;
}

// messages
.colour-warning {
  background: $colour-warning;
}
.colour-failure {
  background: $colour-failure;
}
.colour-success {
  background: $colour-success;
}
.colour-info {
  background: $colour-info;
}

// Data visualisation
.colour-coyote-brown {
  background: $colour-coyote-brown;
}
.colour-outer-space {
  background: $colour-outer-space;
}

.box-colour {
  vertical-align: top;
  display: inline-block;
  width: 9em;
  height: 9em;
  margin-bottom: 4em;
  font-size: $small-font-size;
  &:before {
    content: attr(data-name);
    color: $colour-yankees-blue;
    margin-top: 10em;
    display: block;
    text-align: center;
  }
}
