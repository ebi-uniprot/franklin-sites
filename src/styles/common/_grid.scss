@import '../settings';
@import '../colours';
@import '../mixins';

$grid-name: 'uniprot-grid';
$gap: 1rem;
// Needs to be absolute value, otherwise grid width will change depending on
// where in the page we use it
$max-width: 67rem; // Â±1070px with Lato 16px base as per mockup
$n-columns: 12;

/* uniprot-website grid system */
.#{$grid-name} {
  display: grid;
  gap: $gap; /* or we can implement that differently */
  grid-template-columns: repeat(12, minmax(0, 1fr));

  & > .#{$grid-name}-cell {
    @for $i from 1 through $n-columns {
      &--span-#{$i} {
        grid-column: auto / span $i;
      }
    }

    @for $i from 1 through $n-columns {
      &--offset-#{$i} {
        grid-column-start: $i + 1;
      }
    }
  }

  @each $bp in map-keys($fs-breakpoint-values) {
    @include fs-breakpoints($bp) {
      & > .#{$grid-name}-cell {
        @for $i from 1 through $n-columns {
          &--#{$bp}-span-#{$i} {
            grid-column: auto / span $i;
          }
        }

        @for $i from 1 through $n-columns {
          &--#{$bp}-offset-#{$i} {
            grid-column-start: $i;
          }
        }
      }
    }
  }
}

.#{$grid-name}--centered {
  margin-left: auto;
  width: calc(min(95vw, #{$max-width}));
  margin-right: auto;
}

.#{$grid-name}--with-bleed {
  box-shadow: -$gap 0 var(--grid-background), $gap 0 var(--grid-background);
}
