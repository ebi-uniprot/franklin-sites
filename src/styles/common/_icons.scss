@import '../settings';
@import '../colours';

.icon {
  &--reviewed {
    color: $colour-reviewed;
  }
  &--unreviewed {
    color: $colour-unreviewed;
  }
  &--uniparc {
    color: $colour-uniparc;
  }
  &--reference-proteome {
    color: $colour-reference-proteome;
  }
  &--redundant-excluded-proteome {
    color: $colour-weldon-blue;
  }
  &--community-annotation {
    color: $colour-vivid-cerulean;
  }
}
