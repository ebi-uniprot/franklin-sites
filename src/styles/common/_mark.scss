@import '../colours';

mark {
  font-weight: bold;
  background-color: rgba($colour-medium-turquoise, 0.4);
}
