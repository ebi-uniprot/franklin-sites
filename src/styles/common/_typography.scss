@import '../franklin-settings';
@import '../mixins';

// Font families
@import url('https://fonts.googleapis.com/css?family=Lato:400,700|Source+Sans+Pro:600,700&display=swap');

// Body
html,
body {
  font-family: $fs-base-font-face;
  background-color: $colour-sky-white;

  @include fs-breakpoints('small') {
    font-size: $font-size-small;
  }

  @include fs-breakpoints('medium') {
    font-size: $font-size-medium;
  }

  @include fs-breakpoints('large') {
    font-size: $font-size-large;
  }
}

// Heading tag settings
h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: $fs-headers-font-face;
  color: $colour-yankees-blue;
}

// Typography scale https://type-scale.com/
// 1.250 "Major third"

.x-huge {
  font-size: 3.052rem;
  font-weight: $fs-font-weight-bold;
}

h1,
.huge {
  font-size: 2.441rem;
  font-weight: $fs-font-weight-semi-bold;
}
h2,
.big {
  font-size: 1.953rem;
  font-weight: $fs-font-weight-semi-bold;
}
h3,
.medium {
  font-size: 1.563rem;
  font-weight: $fs-font-weight-semi-bold;
}
h4,
.small {
  font-size: 1.25rem;
  font-weight: $fs-font-weight-semi-bold;
}
h5,
.tiny {
  font-size: 1rem;
  font-weight: $fs-font-weight-semi-bold;
}
// Should not use h6
h6 {
  font-size: 1rem;
  font-weight: $fs-font-weight-regular;
}

p {
  margin-bottom: 1.875rem;
}

ul,
ol {
  margin-left: 1.25rem;
}

ul.no-bullet,
ol.no-bullet {
  list-style: none;
  margin-left: 0;
}

ul,
ol,
dl {
  margin-bottom: 1rem;
  list-style-position: outside;
  line-height: 1.6;
}

ul {
  list-style-type: disc;
}

a {
  color: $colour-link;
  text-decoration: none;
  font-weight: $fs-font-weight-semi-bold;
  &:hover {
    color: lighten($colour-link, 10);
  }
}
