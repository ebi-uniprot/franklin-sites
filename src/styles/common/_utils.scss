@use 'sass:math';
@import '../settings';

$small-bubble: 1.6rem;
$medium-bubble: 3rem;
$large-bubble: 5rem;

$rootPx: 16px;

@function remToPx($rem) {
  @return unitless-calc($rem) * $rootPx;
}

@function pxToRem($px) {
  @return math.div(unitless-calc($px), unitless-calc($rootPx)) + 0px;
}

@function bubbleFontSize($size) {
  @return $size * 0.25 + 0.3333333333;
}

/******************************************************************************\
 *                                                                            *
 * Visually hide any element (mostly text) accessibly.                        *
 * Support includes IE9+                                                      *
 * Source: https://www.scottohara.me/blog/2017/04/14/inclusively-hidden.html  *
 *                                                                            *
 ******************************************************************************/
.visually-hidden:not(:focus):not(:active) {
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}
