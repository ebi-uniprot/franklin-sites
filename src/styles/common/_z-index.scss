$z-index-low: 2000;
$z-index-medium: 5000;
$z-index-high: 9000;
//previous: $z-index-highest: 10000; current: max possible z-index
$z-index-highest: 2147483641;
