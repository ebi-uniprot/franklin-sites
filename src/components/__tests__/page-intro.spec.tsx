import { render, screen } from '@testing-library/react';
import { HeadingLevels } from '../../types/common';

import PageIntro from '../page-intro';

describe('PageIntro component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <PageIntro heading="Title" resultsCount={1000}>
        <div>Some content</div>
      </PageIntro>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with heading level', () => {
    const level = 5;
    const headingLevel = `h${level}` as HeadingLevels;
    render(
      <PageIntro
        heading="Title"
        resultsCount={1000}
        headingLevel={headingLevel}
      >
        <div>Some content</div>
      </PageIntro>
    );
    expect(screen.getByRole('heading', { level })).toBeInTheDocument();
  });

  it('should render 1 singular result', () => {
    render(
      <PageIntro heading="Title" resultsCount={1}>
        <div>Some content</div>
      </PageIntro>
    );
    expect(
      screen.getByRole('heading', { name: /Title 1 result/ })
    ).toBeInTheDocument();
  });

  it('should render title postscript', () => {
    render(
      <PageIntro
        heading="Title"
        resultsCount={1000}
        headingPostscript={<i>from job ID123</i>}
      />
    );
    expect(
      screen.getByRole('heading', {
        name: /Title 1,000 results from job ID123/i,
      })
    ).toBeInTheDocument();
  });
});
