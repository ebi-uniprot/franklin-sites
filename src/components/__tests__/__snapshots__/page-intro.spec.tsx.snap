// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageIntro component should render 1`] = `
<DocumentFragment>
  <div
    class="page-intro"
  >
    <h1
      class=""
    >
      Title
      <small>
         1,000 results 
      </small>
    </h1>
  </div>
</DocumentFragment>
`;
